### PR TITLE
squid service: initial service based on default config

### PR DESCRIPTION
**Pull request by bachp**
###### Motivation for this change

Allow to run squid as a NixOS service.

This is work in progress the configuration is based on the the default configuration to make the service work. Currently only the port can be configured.

I'm planning to allow some more configuration by taking inspiration from [NethServer](https://github.com/NethServer/nethserver-squid).

This is still work in progress but as I'm new to squid and the config is pretty complex I already share my work in case there is a squid expert around wanting to help out.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/networking/squid.nix [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      wantedBy = [ "multi-user.target"];
+      preStart = ''
+        mkdir -p "/var/log/squid"
+        chown squid:squid "/var/log/squid"
+      '';
+      serviceConfig = {
+        Type="forking";
+        PIDFile="/run/squid.pid";
+        PermissionsStartOnly = true;
+        ExecStart  = "${pkgs.squid}/bin/squid -YCs -f ${squidConfig}";
```
Comment (0xABAB) [inline nixos/modules/services/networking/squid.nix]: Use `lib.escapeShellArg`

Comment (bachp) [inline nixos/modules/services/networking/squid.nix]: I don't think anything needs escaping. All arguments are fixed and the config file path is generated. Do I miss something?

**Comment by bachp** [comment]
I did run this for some time now and it works. So for basic proyxing it is enough to do `services.squid.enable = true;`.

I think it can be merged.

**Comment by bachp** [comment]
@bjornfor systemd service description is now starting with capital letter.

**Comment by bachp** [comment]
@fpletz this is no longer WIP

**Review by fpletz — APPROVED** [review]
Thanks! This module is nice and easy so we'll squeeze it into 17.09. :)

**Pull request merged**